common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -286,7 +286,7 @@ function(VideoPlayer, i18n, moment, _) {
      */
     function extractHLSVideoSources(state) {
         return _.filter(state.config.sources, function(source) {
-            return /\.m3u8$/.test(source);
+            return /\.m3u8\??.*$/.test(source);
```

Comment:
This regular expression matches every string containing `.m3u8`, since the question mark is optional, and `.*$` does not pose any restriction whatsoever (it just means that there possibly are further characters until the end of string, which is always true).  So we should either use `/\.m3u8/`, which is equivalent to the current solution, or `/\.m3u8(\?.*)?$, which is what you probably inteded to write.  I'm fine with either.